### ISSUE TITLE
[Python] Generate trailing underscore for fields that shadow reserved keywords

### ISSUE THREAD

**Issue by noahseger** [protocolbuffers/protobuf#17332]
**What language does this apply to?**
Python

**Describe the problem you are trying to solve.**

Fields named, for example `from`, are inaccessible because they shadow the Python keyword.

https://stackoverflow.com/a/30145419

**Describe the solution you'd like**
Additionally (backwards compatible) generate a field named like `from_` in the Python protoc and PYI protoc.

This convention is documented in PEP8:

> single_trailing_underscore_: used by convention to avoid conflicts with Python keyword, e.g.

https://peps.python.org/pep-0008/#descriptive-naming-styles

**Describe alternatives you've considered**
`getattr`/`setattr` hacks.

**Additional context**

**Comment by honglooker** [comment]
FYI @anandolee

**Comment by noahseger** [comment]
I think this should remain active 🙏

**Comment by noahseger** [comment]
🙏

**Comment by mkruskal-google** [comment]
Note, from 2009 in [python/google/protobuf/internal/python_message.py](https://github.com/protocolbuffers/protobuf/blob/main/python/google/protobuf/internal/python_message.py)
```
  # Kenton says:  The above is a BAD IDEA.  People rely on being able to use
  #   getattr() and setattr() to reflectively manipulate field values.  If we
  #   rename the properties, then every such user has to also make sure to apply
  #   the same transformation.  Note that currently if you name a field "yield",
  #   you can still access it just fine using getattr/setattr -- it's not even
  #   that cumbersome to do so.
  # TODO(kenton):  Remove this method entirely if/when everyone agrees with my
  #   position.
```

I'm not quite sure I buy this argument, but it is definitely a major breaking change to do this.  We could probably consider it for a future edition to make it opt-in though.  I will add it to our (quite long) list of ideas for future features

**Comment by noahseger** [comment]
Yes, I agree that the canonical field names should remain accessible via `getattr`/`setattr`. I just think that _additionally_ we can add some protoc option that generates PEP-0008 style shadowed name properties when you opt in so that nothing breaks. What do you think?

**Comment by mkruskal-google** [comment]
We're generally extremely wary of adding new protoc options, because those can *easily* blow up.  It becomes incredibly difficult to test every possible combination and it creates divergent ecosystems where shared libraries can't really support every setting you use to produce gencode.

We've been actively trying to reduce protoc flags, and I think the only realistic way to roll this out that fits into our current evolution model is with a new python editions feature.  You could set `option features.(pb.python).escape_keyword_fields = true` in your .proto file, and protoc would add the underscore.  We could even have a tristate feature where an intermediate value generates name properties to allow for smooth migrations.

I've added this to our list to consider/design, but it seems like a low priority quality of life improvement.  We'd probably wait until we were making similar changes elsewhere to roll out something like this.

**Comment by anandolee** [comment]
keywords are support in python by setattr/getattr:
https://github.com/protocolbuffers/protobuf/blob/main/python/google/protobuf/internal/keywords_test.py
